examples/async-sqlite/src/lib.rs

```diff
@@ -135,7 +134,7 @@ impl Database {
         let name = cx.argument::<JsString>(0)?.value(&mut cx);
 
         // Get the `this` value as a `JsBox<Database>`
-        let db = cx.this().downcast_or_throw::<JsBox<Database>, _>(&mut cx)?;
+        let db = cx.this::<JsBox<Database>>()?;
```

Comment:
General specialization would be awesome here. We could have a blanket impl for downcasting `T`, but a specialized case for when `T` implements `Value`.

This would allow you to skip the `JsBox` here and assume it from an out param.